src/sw.js

```diff
@@ -141,19 +128,53 @@ const navigationNormalizationPlugin = {
 
 precacheAndRoute(self.__WB_MANIFEST);
 
+// HTML caching strategy
+const htmlStrategy = new StaleWhileRevalidate({
+  cacheName: 'pages-cache',
+  plugins: [navigationNormalizationPlugin],
+});
+
+/**
+ * HTML Handler
+ * @param {object} param0 - Workbox Handler
+ * @return {object} - Either a redirection response or a Workbox response
+ */
+async function htmlHandler({ event }) {
+  const lang = await preferences.get('lang');
+  const { request } = event;
+
+  const urlLang = request.url.replace(self.location.origin, '').split('/');
+  const currentLang = urlLang[1];
+  const isALanguage = languages.includes(currentLang);
+  const isRightLanguage = currentLang !== lang;
+  const shouldRefresh = new URL(request.url).searchParams.get('locale_fallback') !== 'true';
+
+  if (isALanguage && isRightLanguage && shouldRefresh) {
+    console.log('⇒ Redirecting');
+    urlLang[1] = lang;
+    const redirectURL = `${self.location.origin}${urlLang.join('/')}`;
+    return Response.redirect(redirectURL, 302);
+  }
+
+  return htmlStrategy.handle({ event, request });
+}
+
+// Handle navigation requests with htmlHandler.
+registerRoute(({ request }) => request.mode === 'navigate', htmlHandler);
+
 // Cache the Google Fonts stylesheets with a stale-while-revalidate strategy.
 registerRoute(
-  /^https:\/\/fonts\.googleapis\.com/,
+  ({ request }) => request.destination === 'style',
```

Comment:
This is a new v5 way of doing Google stylesheet/font caching? I think I copied this (and the other) from the current docs.

Reply:
It worked in v4 as well: https://developers.google.com/web/tools/workbox/guides/common-recipes#cache_resources_based_on_resource_type

You could go back to using `RegExp`s but for something that's "boilerplate"-y, I thought it would be more useful to route based on type of subresource instead of URL patterns. That way if folks are using a different, e.g., web font provider, the code will still work.

Reply:
I didn't know that! That's much better. Thanks!